drivers/src/pmu.rs

```diff
@@ -50,19 +52,50 @@ impl PmuInfo {
         Ok(ControlFlow::Continue(()))
     }
 
-    /// Initializes the global PmuInfo structure with the information obtained from the platform SBI.
-    pub fn init() -> SbiResult<()> {
-        let mut pmu_info = PmuInfo::new();
+    fn get_delegated_counters(&mut self) -> SbiResult<()> {
+        let delegated_counters = CSR.smcounteren.get();
+        // Always ignore index 1 (TIME)
+        self.valid_hw_counters_mask = delegated_counters & !0x2;
```

Comment:
Changed, but still need to invert the mask: `delegated_counters & !(1 << (CSR_TIME - CSR_CYCLE)`. Sorry, @atishp04, they decided to use "!" instead of "~" in Rust for bit-wise negation :).